trestle/core/markdown/control_markdown_node.py

```diff
@@ -239,6 +236,126 @@ def get_control_guidance(self) -> Optional[ControlMarkdownNode]:
         """Get control guidance node."""
         return self.get_node_for_key('## Control Guidance')
 
+    def get_by_id_parts(self) -> Dict[str, List[common.Part]]:
+        """Get by id editable parts."""
+        part_id_to_parts_map = {}
+        for node_key in as_filtered_list(self.content.subnodes_keys, lambda k: re.match(const.PART_REGEX, k.lower())):
+            # A by id part section
+            part_node = self.get_node_for_key(node_key)
+            if not part_node.content.a_part:
+                raise TrestleError(f'Error no part was found in section {part_node.key}.')
+            if part_node.content.by_id_name not in part_id_to_parts_map:
+                part_id_to_parts_map[part_node.content.by_id_name] = []
+            for subpart in part_node.subnodes:
+                # We only care about subsections of ## Part section
+                if subpart.content.a_part:
+                    part_id_to_parts_map[part_node.content.by_id_name].append(subpart.content.a_part)
+
+        return part_id_to_parts_map
+
+    def get_editable_parts_and_subparts(self) -> Optional[List[common.Part]]:
+        """Get editable parts and subparts if exist."""
+        by_id_parts = self.get_by_id_parts()
+        exclude_parts_id = list(by_id_parts.keys())
+        editable_node = self.get_node_for_key(f'# {const.EDITABLE_CONTENT}')
+        parts = self._get_subparts(editable_node, exclude_parts_id)
+        return parts
+
+    def _get_subparts(self,
+                      control_node: ControlMarkdownNode,
+                      exclude_ids: List[str] = None) -> Optional[List[common.Part]]:
+        """Get subparts of the control part node if exists."""
+        if not control_node:
+            raise TrestleError('No control node was provided to extract subparts.')
```

Comment:
Is there a way to provide more context for this error so the user knows where the problem is?  Perhaps the check needs to be before this call.